src/with_clause_parser.h

```diff
@@ -0,0 +1,30 @@
+/*
+ * This file and its contents are licensed under the Apache License 2.0.
+ * Please see the included NOTICE for copyright information and
+ * LICENSE-APACHE for a copy of the license.
+ */
+#include <postgres.h>
+#include <c.h>
+
+#include <nodes/parsenodes.h>
+
+/* find the length of a statically sized array */
+#define TS_ARRAY_LEN(array) (sizeof(array) / sizeof(*array))
+
+typedef struct WithClauseArg
+{
+	const char *arg_name;
+	Oid type_id;
+	Datum default_val;
+} WithClauseArg;
+
+typedef struct WithClauseParse {
```

Comment:
should the name here be `WithClauseResult`

Reply:
will change